library/alloc/src/sync/tests.rs

```diff
@@ -140,6 +140,41 @@ fn unwrap_or_drop() {
     assert_eq!(Arc::unwrap_or_drop(x), Some(5));
 }
 
+#[test]
+fn unwrap_or_drop_linked_list() {
+    #[derive(Clone)]
+    struct LinkedList<T>(Option<Arc<Node<T>>>);
+    struct Node<T>(T, Option<Arc<Node<T>>>);
+
+    impl<T> Drop for LinkedList<T> {
+        fn drop(&mut self) {
+            let mut x = self.0.take();
+            while let Some(arc) = x.take() {
+                Arc::unwrap_or_drop(arc).map(|node| x = node.1);
+            }
+        }
+    }
+
+    impl<T> LinkedList<T> {
+        fn push(&mut self, x: T) {
+            self.0 = Some(Arc::new(Node(x, self.0.take())));
+        }
+    }
+
+    use std::thread;
+    for _ in 0..25 {
```

Comment:
Why the loop with `0..25` ?

Reply:
That’s a number that makes it likely enough to hit the race condition while not slowing the test down too much. As I said above, I think this test could be removed entirely.